instrumentation_helpers/lib/opentelemetry/instrumentation_helpers/http/instrumentation_options.rb

```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module InstrumentationHelpers
+    module HTTP
+      # Instrumentation options contains instrumentation helpers for http instrumentation client requests
+      module InstrumentationOptions
+        def self.included(klass)
+          klass.option(:hide_query_params, default: true, validate: :boolean) if klass.respond_to?(:option)
```

Comment:
This feels overly defensive to me. If we want to do this, though, we should `raise` if the class doesn't respond to `option`.

We need a convention for documenting these options, and referencing them from the class that includes them. I don't think we want to copy option documentation - that seems very brittle.

I wonder if there's value in "self documenting" options in a way that'd let you ask (programmatically) both "what options does this instrumentation support?" and "what are the configuration values for this 'instance' of the instrumentation, and which values differ from the default?"

Reply:
The boolean option weirds me out a bit, probably because we had something similar for DB instrumentation that we ended up deprecating in favour of a tristate option supporting obfuscation. I wonder if we're going to end up in the same place with this.

Reply:
spent a bit of time thinking through some ideas this morning. I'm starting to think this is going to be a separate issue/PR from the request attribute http helpers, and we should scope down this pr to only address the request attribute helper stuff.

tl;dr is I am reviewing how dd-trace-rb instrumentation handles organizing [settings](https://github.com/DataDog/dd-trace-rb/blob/e391d2eb64d3c6151a4bdd2710c6a8c7c1d57457/lib/ddtrace/contrib/active_record/configuration/settings.rb#L11), as we were able to programmatically expose setting info via [diagnostic logs](https://github.com/DataDog/dd-trace-rb/blob/e391d2eb64d3c6151a4bdd2710c6a8c7c1d57457/lib/ddtrace/diagnostics/environment_logger.rb#L262) in a way that I think is similar to the needs for `self.documenting` instrumentation, and I think the approach taken there could be applied here.

also Apologies if my terminology or pseudo code is a bit off.

Anyway, I think allowing programmatic control over instrumentation options involves refactoring (or at least further review as it's been awhile) of `Instrumentation::Base` class,  and likely an introduction of something like a `Instrumentation::Base::Settings` or `Instrumentation::Base::Configuration` class, which specific instrumentation can inherit from, and augment. This class would basically encapsulate all the various `Instrumentation::Base.option(option_details)` invocations sprinkled in the Instrumentation class, and be a single source of truth for what keys are available for an instrumentation's options. Ideally it would expose some helper methods that could, for example, build a hash of all the instrumentations `option_name / default / validate` configuration settings, and then instrumentations like say `OpenTelemetry::Instrumentation::Bunny` could simply expose method like `default_configuration`, that could gather information about the settings class (or instantiate a new settings class instance, etc) without us necessarily having to create an instance of the entire instrumentation.

I haven't really fully spec'd this out but, i think for this PR, i'd rather keep the old/somewhat hacky pattern of just manually adding `.option(:hide_query_params,...` method calls to all the relevant instrumentation, and scoping this PR to only be a helper for the http request attributes.

As this has been sitting on my plate for awhile, and the other comments are all pretty reasonable and straightforward, it probably doesn't make sense to tie up releasing that work with a chunky refactor of instrumentation.

wdyt?

Reply:
been working through this a bit more...i think i have something that is maybe managable to include in this pr, gonna try to push it up by eod